### DISCUSSION TITLE
how to reproduce migration failure

### DISCUSSION THREAD

**Discussion by thilllon** [prisma/prisma#17018] [Q&A]
We are currently building a CI test pipeline that includes down-migration process after Prisma migration failure. https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/generating-down-migrations#how-to-generate-and-run-down-migrations

But I can't quite figure out how to reproduce the situation where it fails on purpose. Is there any easy, simple way to make migration fail?

Thanks.

(To use broken SQL is not a good idea because we suppose that we are going to use migrations(*.sql) which are auto-generated by Prisma that are not broken. So it is not suitable for what we expect from the CI.)

**Comment by pantharshit00** [comment]
I guess you can try to violate a key constraint or use non existent columns in the migration SQL to test this out. Let me know if these will work here for you. You can always edit and add `INSERT` or any other DQL statement in the file as well along with the DDL queries.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
